### PR TITLE
Update lightblue-client error handling. It throws now a LightblueExce…

### PR DESCRIPTION
**Pull request by ykoer**
…ption containing the Errors and DataErrors

### CODE REVIEW

**Comment by bserdar** [comment]
Few things:
 1) Error.java needs cleanup. No need for logging, threadlocals, static
get() methods, constructors push/pup, and toJson()
 2) Add a LightblueResponse instance to the LightblueException, instead of
just the errors. Otherwise, there is no way to get the response. errorCodes
set can be initialized lazily.
 3) There is no need for JsonObject class. And there is no need for toJson
methods. Errors are already parsed from json.

File: core/src/main/java/com/redhat/lightblue/client/model/Error.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * client as an indicator of where the error happened.
+ * 
+ */
+public final class Error extends RuntimeException {
+
+    private static final long serialVersionUID = 1L;
+
+    public static final char DELIMITER = '/';
+
+    private final ArrayDeque<String> context;
```
Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/model/Error.java]: String context should be enough.

File: core/src/main/java/com/redhat/lightblue/client/model/Error.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    public String getErrorCode() {
+        return errorCode;
+    }
+
+    public String getMsg() {
+        return msg;
+    }
+
+    public void pushContext(String context) {
```
Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/model/Error.java]: No need for this once context is string

File: core/src/main/java/com/redhat/lightblue/client/model/Error.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+            }
+            x = ((ObjectNode) node).get("msg");
+            if (x != null) {
+                m = x.asText();
+            }
+
+            Error ret = new Error(e, m);
+
+            x = ((ObjectNode) node).get("context");
+            if (x != null) {
```
Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/model/Error.java]: And you can get rid of this as well. Context is simply a string for the client

File: core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    public void setLightblueResponse(DefaultLightblueResponse lightblueResponse) {
+        this.lightblueResponse = lightblueResponse;
+    }
+
+    public boolean exists(String errorCode) {
+        return errorCodes.contains(errorCode);
+    }
+
+    public Set<String> getErrorCodes() {
+        return errorCodes;
```
Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java]: initialize error codes here, instead of in the constructor

Comment (ykoer) [inline core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java]: I would then have to initialize the error codes in "exists" method as well, which I don't like.

Comment (bserdar) [inline core/src/main/java/com/redhat/lightblue/client/response/LightblueException.java]: You could do getErrorCodes().contains(errorCode)

Not a big deal though.

**Pull request merged**